src/pyconverter/xml2py/writer.py

```diff
@@ -270,115 +408,257 @@ def write_source(
+                continue
+            python_name = name_map[initial_command_name]
+            path = os.path.join(library_path, f"{python_name}.py")
+            python_method = command_obj.to_python(custom_functions)
+            try:
+                exec(python_method)
+                with open(path, "w", encoding="utf-8") as fid:
+                    fid.write(f"{python_method}\n")
+            except Exception as e:
+                raise RuntimeError(f"Failed to execute {python_name}.py") from e
 
-    mod_file = os.path.join(cmd_path, "__init__.py")
-    with open(mod_file, "w") as fid:
-        for ans_name in commands:
-            if ans_name in SKIP_XML:
+    else:
+        import subprocess
+
+        package_structure = {}
+        all_commands = []
+        specific_classes = get_config_data_value(config_path, "specific_classes")
+        for command in tqdm(command_map.values(), desc="Writing commands"):
+            if command.name in SKIP_XML or command.group is None:
                 continue
-            cmd_name = cmd_map[ans_name]
-            fid.write(f"from .{cmd_name} import *\n")
-        fid.write("try:\n")
-        fid.write("    import importlib.metadata as importlib_metadata\n")
-        fid.write("except ModuleNotFoundError:\n")
-        fid.write("    import importlib_metadata\n\n")
-        fid.write("__version__ = importlib_metadata.version(__name__.replace('.', '-'))\n")
-        fid.write('"""PyConverter-GeneratedCommands version."""\n')
 
-    print(f"Commands written to {cmd_path}")
+            module_name, initial_class_name, module_path = get_module_info(library_path, command)
+
+            # Create the module folder and structure if it doesn't exist yet
+            if not os.path.isdir(module_path):
+                os.makedirs(module_path)
+                package_structure[module_name] = {}
+
+            # Check whether the class name needs to follow a specific rule
+            if initial_class_name in specific_classes.keys():
+                initial_class_name = specific_classes[initial_class_name]
+
+            class_name, file_name, file_path = get_class_info(initial_class_name, module_path)
+
+            # Create the class file and structure if it doesn't exist yet
+            if not os.path.isfile(file_path):
+                class_structure = []
+                with open(file_path, "w", encoding="utf-8") as fid:
+                    fid.write(f"class {class_name}:\n")
+            else:
+                # Get the class structure
+                class_structure = package_structure[module_name][file_name][1]
+            class_structure.append(command.py_name)
+
+            package_structure[module_name][file_name] = [class_name, class_structure]
+
+            # Write the Python method to the class file
+            with open(file_path, "a", encoding="utf-8") as fid:
+                python_method = command.to_python(custom_functions, indent="    ")
```

Comment:
```suggestion
                python_method = command.to_python(custom_functions, indent=4*" ")
```

Reply:
this is just to make more clear how many spaces there is.

Reply:
Good approach, thanks!